store/postgres/job_deployment_repository.go

```diff
@@ -0,0 +1,109 @@
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/optimus/store"
+)
+
+type JobDeployment struct {
+	ID        uuid.UUID `gorm:"not null" json:"id"`
+	ProjectID uuid.UUID `gorm:"not null" json:"project_id"`
+	Project   Project   `gorm:"foreignKey:ProjectID"`
+
+	Status  string `gorm:"not null"`
```

Comment:
Please mark the default value as null so that it returns error in case of empty status otherwise it can insert empty string.
Gorm attribute :
```
Status  string `gorm:"not null;default:null"`
```

Reply:
updated this.